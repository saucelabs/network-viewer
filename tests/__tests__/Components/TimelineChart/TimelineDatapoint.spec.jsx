import React from 'react';
import { mount } from 'enzyme';

import TimelineChartDatapoint from './../../../../src/Components/TimelineChart/TimelineDatapoint';

describe('TimelineChartDatapoint', () => {
  it('renders without crashing', () => {
    const element = mount(
      <svg>
        <TimelineChartDatapoint
          cx={234}
          index={5}
          maxTime={17000}
          payload={{
            timings: {
              startTime: 1200,
            },
          }}
        />
      </svg>,
    );
    expect(element).toMatchSnapshot();
  });
});
