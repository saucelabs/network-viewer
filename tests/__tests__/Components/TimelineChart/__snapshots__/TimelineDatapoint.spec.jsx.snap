// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimelineChartDatapoint renders without crashing 1`] = `
<svg>
  <TimelineDatapoint
    cx={234}
    cy={0}
    index={5}
    maxTime={17000}
    payload={
      Object {
        "timings": Object {
          "startTime": 1200,
        },
      }
    }
  >
    <g>
      <rect
        fill="#ccc"
        height={2}
        key="queueing"
        width="0%"
        x="7.0588235294117645%"
        y={55}
      />
      <rect
        fill="#A1000C"
        height={2}
        key="blocked"
        width="0%"
        x="14.117647058823529%"
        y={55}
      />
      <rect
        fill="#DCC9E5"
        height={2}
        key="dns"
        width="0%"
        x="21.176470588235293%"
        y={55}
      />
      <rect
        fill="#E78057"
        height={2}
        key="ssl"
        width="0%"
        x="28.235294117647058%"
        y={55}
      />
      <rect
        fill="#DB8553"
        height={2}
        key="connect"
        width="0%"
        x="35.294117647058826%"
        y={55}
      />
      <rect
        fill="#3C96C4"
        height={2}
        key="send"
        width="0%"
        x="42.352941176470594%"
        y={55}
      />
      <rect
        fill="#7CA0BF"
        height={2}
        key="wait"
        width="0%"
        x="49.41176470588236%"
        y={55}
      />
      <rect
        fill="#65B955"
        height={2}
        key="receive"
        width="0%"
        x="56.47058823529413%"
        y={55}
      />
    </g>
  </TimelineDatapoint>
</svg>
`;
